src/backend/cpu/cholesky.cpp

```diff
@@ -70,13 +71,7 @@ int cholesky_inplace(Array<T> &in, const bool is_upper)
     if(is_upper)
         uplo = 'U';
 
-    int info = 0;
-    auto func = [&] (int& info, Array<T>& in) {
-        info = potrf_func<T>()(AF_LAPACK_COL_MAJOR, uplo, N, in.get(), in.strides()[1]);
-    };
-
-    getQueue().enqueue(func, info, in);
-    getQueue().sync();
+    int info = potrf_func<T>()(AF_LAPACK_COL_MAJOR, uplo, N, in.get(), in.strides()[1]);
```

Comment:
This work will be performed on the calling thread and all other functions will be called on the queue thread. While this is a synchronous function, it might be beneficial to keep all work on the queue thread for consistency.